lib/tapp.hoon

```diff
@@ -390,12 +488,19 @@
       :-  ~
       %-  from-form:eval:tapp-async
       ^-  form:tapp-async
-      ?-  -.u.next
-        %init  ~(handle-init handler bowl app-state)
-        %poke  (~(handle-poke handler bowl app-state) +.u.next)
-        %peer  (~(handle-peer handler bowl app-state) +.u.next)
-        %diff  (~(handle-diff handler bowl app-state) +.u.next)
-        %take  (~(handle-take handler bowl app-state) +.u.next)
+      =/  out
+        %-  mule  |.
```

Comment:
This is very good.  We still need to think about non-deterministic errors at some point.